tests/testthat/test-validate.R

```diff
@@ -0,0 +1,322 @@
+  template_invalid_type <- c(
+    "#' @title Test Component",
+    "#' @description Test description", 
+    "#' @type invalid_type",
+    "#' @code",
+    ".self"
+  )
+  
+  expect_error(
+    validate_template(template_invalid_type),
+    "@type must be one of"
+  )
+})
+
+test_that("validate_template accepts valid @type values", {
+  valid_types <- c("predecessor", "derivation", "row")
+  
+  for (type in valid_types) {
+    template <- c(
+      "#' @title Test Component",
+      "#' @description Test description",
+      paste("#' @type", type),
+      "#' @code", 
+      ".self"
+    )
+    
+    expect_no_error(validate_template(template))
+  }
+})
+
+test_that("validate_template requires non-empty @code section", {
+  template_empty_code <- c(
+    "#' @title Test Component",
+    "#' @description Test description",
+    "#' @type derivation",
+    "#' @code"
+  )
+  
+  expect_error(
+    validate_template(template_empty_code),
+    "@code.*section cannot be empty"
+  )
+  
+  template_whitespace_code <- c(
+    "#' @title Test Component", 
+    "#' @description Test description",
+    "#' @type derivation",
+    "#' @code",
+    "   ",
+    "  "
+  )
+  
+  expect_error(
+    validate_template(template_whitespace_code),
+    "@code.*section cannot be empty"
+  )
+})
+
+test_that("@param missing description", {
+  # Missing description
+  template_bad_param <- c(
+    "#' @title Test Component",
+    "#' @description Test description", 
+    "#' @param variable",
+    "#' @type derivation",
+    "#' @code",
+    ".self"
+  )
+  expect_error(
+    mighty_component$new(template_bad_param, "A"),
+    "Invalid.*@param.*Must have both name and description"
+  )
+})
+
+test_that("multi-line @param description works", {
+  # Missing description
+  template_bad_param <- c(
+    "#' @title Test Component",
+    "#' @description Test description", 
+    "#' @param variable long multi line description that
+    #' goes",
+    "#' @type derivation",
+    "#' @code",
+    ".self"
+  )
+
+  expect_no_error(
+    mighty_component$new(template_bad_param, "A"))
+})
+
+test_that("@depends tags needs both domain and column", {
+  # Missing column
+  template_bad_depends <- c(
+    "#' @title Test Component",
+    "#' @description Test description",
+    "#' @depends .self",
+    "#' @type derivation",
+    "#' @code", 
+    ".self"
+  )
+  
+  expect_error(
+    validate_template(template_bad_depends),
+    "Invalid.*@depends.*Must have both domain and column"
+  )
+  
+  # Valid depends format
+  template_valid_depends <- c(
+    "#' @title Test Component",
+    "#' @description Test description",
+    "#' @depends .self USUBJID",
+    "#' @type derivation",
+    "#' @code",
+    ".self"
+  )
+  
+  expect_no_error(validate_template(template_valid_depends))
+})
+
+test_that("validate_template validates @outputs tags", {
+  template_valid_output <- c(
+    "#' @title Test Component",
+    "#' @description Test multile",
```

Comment:
Typo in test data: 'multile' should be 'multiple'
```suggestion
    "#' @description Test multiple",
```